### ISSUE TITLE
Schema validation error

### ISSUE THREAD

**Issue by SBurwash** [z3z1ma/target-bigquery#11]
I received an error in my tap regarding schema validation: 

```
2023-01-05T15:28:01.267692Z [info     ]   File "/project/.meltano/loaders/target-bigquery/venv/lib/python3.8/site-packages/jsonschema/validators.py", line 314, in validate cmd_type=elb consumer=True name=target-bigquery--hibob-finance producer=False stdio=stderr string_id=target-bigquery--hibob-finance
2023-01-05T15:28:01.267783Z [info     ]     raise error                cmd_type=elb consumer=True name=target-bigquery--hibob-finance producer=False stdio=stderr string_id=target-bigquery--hibob-finance
2023-01-05T15:28:01.267871Z [info     ] jsonschema.exceptions.ValidationError: 100 is not of type 'string', 'null' cmd_type=elb consumer=True name=target-bigquery--hibob-finance producer=False stdio=stderr string_id=target-bigquery--hibob-finance
2023-01-05T15:28:01.267960Z [info     ]                                cmd_type=elb consumer=True name=target-bigquery--hibob-finance producer=False stdio=stderr string_id=target-bigquery--hibob-finance
2023-01-05T15:28:01.268054Z [info     ] Failed validating 'type' in schema['properties']['payroll']['properties']['variablePayment']['items']['properties']['individualPercent']: cmd_type=elb consumer=True name=target-bigquery--hibob-finance producer=False stdio=stderr string_id=target-bigquery--hibob-finance
2023-01-05T15:28:01.268144Z [info     ]     {'type': ['string', 'null']} cmd_type=elb consumer=True name=target-bigquery--hibob-finance producer=False stdio=stderr string_id=target-bigquery--hibob-finance
2023-01-05T15:28:01.268232Z [info     ]                                cmd_type=elb consumer=True name=target-bigquery--hibob-finance producer=False stdio=stderr string_id=target-bigquery--hibob-finance
2023-01-05T15:28:01.268315Z [info     ] On instance['payroll']['variablePayment'][4]['individualPercent']: cmd_type=elb consumer=True name=target-bigquery--hibob-finance producer=False stdio=stderr string_id=target-bigquery--hibob-finance
2023-01-05T15:28:01.268404Z [info     ]     100      
```

where I had misidentified a type for a column. While this is not problematic per say and makes me change my schema, it was my understanding that this target did not perform schema validation, rather going the path of letting the user perform their own validation in their warehouse using DBT.

Was this a misunderstanding on my part?

**Comment by z3z1ma** [comment]
This function signature here in the SDK is what is causing that:

```python
    def _validate_and_parse(self, record: dict) -> dict:
```

___

To answer you though,

> rather going the path of letting the user perform their own validation in their warehouse using DBT.

That idea manifests in the **fixed schema** load pattern where we put all the data in an unstructured JSON column. The fact the schema is static in that pattern means it will always pass validation. The denormalized pattern using the actual schema will pas through the SDK plumbing as usual. I think its more of a feature than an issue. I cannot say for certain disabling that wouldn't break existing loads that rely on it.

**Comment by z3z1ma** [comment]
Closing this as no new activity and its not an error but expected behavior. 
Please feel free to raise a discussion instead if there is any opinion on this factoring in my comments above.

**Comment by z3z1ma** [comment]
@SBurwash 

I did some deeper digging into the SDK code and I don't think type conformance happens in `_validate_and_parse` so I went ahead and made it a no-op in the recent release (https://pypi.org/project/z3-target-bigquery/). We should let GCP handle what it can handle and the in flight validation is useless to me as BigQuery will tell us if there is a problem. My hesitation before was that I though this method was coercing types which could have made it a feature that was depended on in some edge-case but it does not seem to do any coercion.
